.teamcity/src/main/kotlin/configurations/Gradleception.kt

```diff
@@ -20,23 +26,61 @@ class Gradleception(model: CIBuildModel, stage: Stage) : BaseGradleBuildType(sta
         javaCrash = false
     }
 
+    params {
+        // Override the default commit id so the build steps produce reproducible distribution
+        param("env.BUILD_COMMIT_ID", "HEAD")
+    }
+
+    /*
+     To avoid unnecessary rerun, what we do here is a bit complicated:
+
+     1. Build a Gradle distribution with a fixed timestamp and hash it, but never use this distribution.
+     2. Build a Gradle distribution with this hash as a version + fixed timestamp -> dogfood-first
+     3. Build a Gradle distribution using dogfood-first with the "hash of dogfood-first" as a version + fixed timestamp different from the one above -> dogfood-second
```

Comment:
```suggestion
     3. Build a Gradle distribution using dogfood-first with this hash as a version + fixed timestamp different from the one above -> dogfood-second
```

The comment isn't accurate any more.